projects/components/src/datagrid/datagrid.component.ts

```diff
@@ -827,9 +827,11 @@ export class DatagridComponent<R extends B, B = any> implements OnInit, AfterVie
         if (this.datagrid.selection.currentSingle) {
             this._datagridSelection = [this.datagrid.selection.currentSingle];
         } else if (this.datagrid.selection.current && this.datagrid.selection.current.length) {
-            this._datagridSelection = [...this.datagrid.selection.current];
+            this._datagridSelection = this.datagrid.selection.current;
         } else {
```

Comment:
I would write this in the pattern of the previous rows, i.e.:
} else if (this._datagridSelection.length) {
}

and add a comment - that if there is no selection from the clarity we should clear our (in case there is such) without creating a new reference.

Reply:
Why write it in the pattern of the previous rows? If we do that, once the selection length gets to 0, it would be impossible to select another entity, right?